internal/controller/authproxyworkload_controller_test.go

```diff
@@ -232,6 +233,92 @@ func TestReconcileState33(t *testing.T) {
 
 }
 
+func TestReconcileDeleteUpdatesWorkload(t *testing.T) {
+	resource := testhelpers.BuildAuthProxyWorkload(types.NamespacedName{
+		Namespace: "default",
+		Name:      "test",
+	}, "project:region:db")
+	resource.Generation = 1
```

Comment:
If we end up doing this manual adjusting of a resource, we might think about how to wrap this up in a more readable function.